src/Symfony/Component/ExpressionLanguage/ExpressionFunction.php

```diff
@@ -42,8 +42,8 @@ class ExpressionFunction
     public function __construct(string $name, callable $compiler, callable $evaluator)
     {
         $this->name = $name;
-        $this->compiler = $compiler;
-        $this->evaluator = $evaluator;
+        $this->compiler = $compiler instanceof \Closure ? $compiler : \Closure::fromCallable($compiler);
+        $this->evaluator = $evaluator instanceof \Closure ? $evaluator : \Closure::fromCallable($evaluator);
```

Comment:
I'm not 100% sure about this change. `ExpressionFunction` does have a getter for each of both properties (why though?). If we expect those getters to return exactly the callables that were passed to the construtor, that expectation wouldn't be met anymore.

Reply:
those getters are called in \Symfony\Component\ExpressionLanguage\ExpressionLanguage::addFunction.
what's the value of wrapping the callables in a closure? I guess we could add the return type to the getters then?

Reply:
> what's the value of wrapping the callables in a closure?

`callable` cannot be used as a property type. Also, when calling the callable multiple times, a closure should be slightly faster.

> I guess we could add the return type to the getters then?

I think so, yes.

Reply:
I'm fine with the change, but let's do it on 5.4, with the `@return \Closure` annotations

Reply:
#42127

Reply:
Backport has been merged, PR is rebased.